### PR TITLE
feat: classify proto type data as secret, sensitive or public

### PR DESCRIPTION
**Pull request by jimlambrt**

### CODE REVIEW

File: github.com.hashicorp.go.kms.wrapping.v2.types.proto [hunk 1]
```diff
@@ -19,25 +19,31 @@ message WrapperConfig {
 // decryption in an envelope fashion
 message EnvelopeInfo {
   // Ciphertext is the ciphertext from the envelope
+  // @gotags: `class:"public"`
```
Comment (johanbrandhorst) [inline github.com.hashicorp.go.kms.wrapping.v2.types.proto:22]: Nit: I found that the backticks are optional, and I kinda prefer this without them? In Boundary we're not using either consistently, but I think we can do it here .
```suggestion
  // @gotags: class:"public"
```

File: tools/tools.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+// $ make tools
+// or
+// $ go generate -tags tools tools/tools.go
+
+package tools
+
+// NOTE: This must not be indented, so to stop goimports from trying to be
+// helpful, it's separated out from the import block below. Please try to keep
+// them in the same order.
+//go:generate go install github.com/favadi/protoc-go-inject-tag
```
Comment (johanbrandhorst) [inline tools/tools.go:20]: Can we not just use the same way of versioning this as we do buf? I.e. just a `go install` in the Makefile?

File: github.com.hashicorp.go.kms.wrapping.v2.types.proto (hunk 1 above)
Comment (jimlambrt) [reply github.com.hashicorp.go.kms.wrapping.v2.types.proto]: ty

File: Makefile [hunk 3]
```diff
@@ -4,4 +4,13 @@ proto:
...
 	
+	# inject classification tags (see: https://github.com/hashicorp/go-eventlogger/tree/main/filters/encrypt)
+	@protoc-go-inject-tag -input=./github.com.hashicorp.go.kms.wrapping.v2.types.pb.go
+	
 .PHONY: proto
+
+
+.PHONY: tools
+tools: 
+	go install github.com/favadi/protoc-go-inject-tag
```
Comment (johanbrandhorst) [inline Makefile:15]: Does this work? I think we should explicitly version it still.

Comment (jimlambrt) [reply Makefile]: yes, it works.  happy to pin it.

File: .github/workflows/go.yml [hunk 4]
```diff
@@ -161,6 +161,13 @@ jobs:
         with:
           # The 'main' branch of the GitHub repository that defines the module.
           against: "https://github.com/${GITHUB_REPOSITORY}.git#branch=main"
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v4
+        with:
+          go-version-file: go.mod    
+      - name: install go-inject-tag dependency
+        run: |
+          go install github.com/favadi/protoc-go-inject-tag@v1.4.0
```
Comment (johanbrandhorst) [inline .github/workflows/go.yml:170]: Shouldn't this just be `make tools`?

Comment (jimlambrt) [reply .github/workflows/go.yml:170]: Nope.

**Review by johanbrandhorst — APPROVED** [review]

**Pull request merged**